eZ/Publish/Core/Repository/Repository.php

```diff
@@ -260,21 +269,28 @@ public function __construct(
             $this->serviceSettings['language']['languages'] = $this->serviceSettings['languages'];
         }
 
-        if ($user !== null) {
-            $this->setCurrentUser($user);
+        if ($user instanceof User) {
+            $this->currentUser = $user;
+            $this->currentUserRef = new UserRef($user->getUserId());
+        } elseif ($user instanceof APIUserRef) {
+            $this->currentUserRef = $user;
+        } else {
+            $this->currentUserRef = new UserRef($this->serviceSettings['user']['anonymousUserID']);
         }
     }
 
     /**
      * Get current user.
      *
+     * @deprecated 5.3.6 and 5.4.3, use {@link getCurrentUserRef}
```

Comment:
idea was to try to avoid people using this when they only need userRef, but indeed it can be rather be inline doc for best practice. (we should have a tag for that maybe)